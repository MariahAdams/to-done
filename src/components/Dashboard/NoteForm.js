import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteForm extends Component {

  state = {
    key: null,
    title: '',
    notes: '',
    completed: false
  };

  static propTypes = {
    note: PropTypes.object,
    onComplete: PropTypes.func.isRequired
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { key, title, notes, completed } = this.state;
    const note = { title, notes, completed };
    if(key) note.key = key;

    this.props.onComplete(note)
      .then(() => {
        if(!key) return;
        this.setState({ title: '', notes: '', completed: false });
      });
  };

  render() {

    return (
      <div>
        <h3>I am the note form</h3>
        <form onSubmit={this.handleSubmit}>
          <label>Title:<input type="text" name="title"></input></label>
          <label>Notes:<textarea rows="4" cols="50" name="notes"></textarea></label>
          <p>
            <button type="submit">Submit</button>
          </p>
        </form>
      </div>
    );
  }
}

export default NoteForm;