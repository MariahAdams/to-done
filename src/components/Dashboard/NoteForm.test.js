import NoteForm from './NoteForm';
import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Note Form', () => {

  it('renders add if no note prop', () => {
    const handleComplete = jest.fn();
    const promise = Promise.resolve();
    handleComplete.mockReturnValueOnce(promise);

    const wrapper = mount(<NoteForm onComplete={handleComplete}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();

    const note = {
      title: 'finish this lab',
      notes: 'you can do it!',
      completed: false
    };

    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: note.title
      }
    });

    wrapper.find('textarea[name="notes"]').simulate('change', {
      target: {
        name: 'notes',
        value: note.notes
      }
    });

    wrapper.find('input[name="completed"]').simulate('change', {
      target: {
        name: 'completed',
        value: note.completed
      }
    });

    wrapper.find('button').simulate('submit');

    const calls = handleComplete.mock.calls;
    expect(calls.length).toBe(1);
    expect(calls[0][0]).toEqual(note);

    return promise
      .then(() => {
        wrapper.update();
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
  });
});