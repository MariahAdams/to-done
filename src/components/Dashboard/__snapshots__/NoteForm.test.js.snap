// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Form renders add if no note prop 1`] = `
<NoteForm
  onComplete={[MockFunction]}
>
  <div>
    <h3>
      I am the note form
    </h3>
    <form
      onSubmit={[Function]}
    >
      <label>
        Title:
        <input
          name="title"
          type="text"
        />
      </label>
      <label>
        Notes:
        <textarea
          cols="50"
          name="notes"
          rows="4"
        />
      </label>
      <p>
        <button
          type="submit"
        >
          Submit
        </button>
      </p>
    </form>
  </div>
</NoteForm>
`;
